tests/lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,422 @@
+        "const bar = { ...foo }",
+        "Object.assign(...foo)",
+        "Object.assign(foo, { bar: baz })"
+    ],
+
+    invalid: [
+        {
+            code: "Object.assign({}, foo)",
+            output: "({...foo})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar' })",
+            output: "({foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, baz, { foo: 'bar' })",
+            output: "({...baz, foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar', baz: 'foo' })",
+            output: "({foo: 'bar', baz: 'foo'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({ foo: 'bar' }, baz)",
+            output: "({foo: 'bar', ...baz})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Many args
+        {
+            code: "Object.assign({ foo: 'bar' }, cats, dogs, trees, birds)",
+            output: "({foo: 'bar', ...cats, ...dogs, ...trees, ...birds})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // TODO: Handle nested Object.assign calls
+        {
+            code:
+                "Object.assign({ foo: 'bar' }, Object.assign({ bar: 'foo' }, baz))",
+            output: "({foo: 'bar', ...Object.assign({ bar: 'foo' }, baz)})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                },
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code:
+                "Object.assign({ foo: 'bar' }, Object.assign({ bar: 'foo' }, Object.assign({}, { superNested: 'butwhy' })))",
+            output: "({foo: 'bar', ...Object.assign({ bar: 'foo' }, Object.assign({}, { superNested: 'butwhy' }))})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
```

Comment:
Could this error (and the ones below) have more information, e.g. line/column numbers?